hs_core/views/resource_folder_hierarchy.py

```diff
@@ -42,6 +42,7 @@ def data_store_structure(request):
     store_path = str(store_path).strip()
     if not store_path:
         return HttpResponse('Bad request - store_path cannot be empty', status=400)
+    # this is federated if warranted.

```

Comment:
@alvacouch Don't quite understand this added comment here. In any case, this function can be refactored a bit to call the list_folder utility function you added.